lib/posix/c_lib_ext/fnmatch.c

```diff
@@ -48,6 +48,159 @@
 
 #define EOS '\0'
 
+#define MATCH_CLASS6(p, a, b, c, d, e, f, g)                                                       \
+	((p)[0] == (a) && (p)[1] == (b) && (p)[2] == (c) && (p)[3] == (d) && (p)[4] == (e) &&      \
+	 (p)[5] == (f) && (p)[6] == (g))
+
+#define MATCH_CLASS7(p, a, b, c, d, e, f, g, h)                                                    \
+	((p)[0] == (a) && (p)[1] == (b) && (p)[2] == (c) && (p)[3] == (d) && (p)[4] == (e) &&      \
+	 (p)[5] == (f) && (p)[6] == (g) && (p)[7] == (h))
+
+enum fnm_char_class {
+	FNM_CC_ALNUM,
+	FNM_CC_ALPHA,
+	FNM_CC_CNTRL,
```

Comment:
Need to include handling for `[:blank:]` as well.
https://pubs.opengroup.org/onlinepubs/9699919799/basedefs/V1_chap09.html#tag_09_03_05

```suggestion
	FNM_CC_BLANK,
	FNM_CC_CNTRL,
```